### PR TITLE
don't assign github teams

### PR DESCRIPTION
**Pull request by BenTheElder**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: changes `dep-approvers` to the OWNERS_ALIASES entry and `approvers` of the dependency management files instead of the github team alias.

the team can still be CCed when necessary, but there's no reason to assign the entire team to any of these changes, especially when they cannot actually approve them.

we should let blunderbluss assign reviewers out of the alias instead, and not the entire team.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: OWNERS
Comment (BenTheElder) [inline OWNERS]: cc @kubernetes/dep-approvers

**Comment by BenTheElder** [comment]
/hold
cc @liggitt

**Comment by liggitt** [comment]
makes sense to me

**Comment by apelisse** [comment]
Thanks!

**Comment by cblecker** [comment]
Can we keep the hold in place till end of day tomorrow? I’m OoO today, and I’d like to ponder how we can do better on this

**Comment by BenTheElder** [comment]
almost EOD #3

File: OWNERS
Comment (cblecker) [inline OWNERS]: So all this does is remove reviewers. The approve plugin doesn't support regex rn.

**Review by cblecker — CHANGES_REQUESTED** [review]
/hold

This won't work.

File: OWNERS
Comment (BenTheElder) [reply OWNERS]: review assignment inspects `approvers`.

Comment (BenTheElder) [reply OWNERS]: we already could not approve everything, and assigning a github team is an ineffectual pattern.

Comment (BenTheElder) [reply OWNERS]: aliases are also public, while github teams are opaque

**Comment by BenTheElder** [comment]
Auto assigning the entire team for review the moment any PR that touches one of these directories is open is not reasonable. I've not seen this anti-pattern anywhere else.

People should be reaching out for approval **after** review. We do not assign root approvers to do initial PR reviews.

This is an incredibly broken workflow (not to mention that @liggitt ultimately has to approve nearly every PR anyhow...)

PRs that could actually use dependency review are not getting it because I have to find them in the sea of broken / WIP .. PRs that touched dependencies in some way when they were opened that aren't even relevant anymore.

(NOTE: We never remove reviewers unless explicitly asked, and you it's impossible to ask prow to remove teams due to github limitations, and I don't have the github permissions to edit reviewers directly)

**Comment by BenTheElder** [comment]
cc current members @dims @sttts @soltysh @thockin @apelisse (still in the github team but not in the OWNERS anymore...) @cblecker 

----

I've set this team to "not watching" myself for now to mitigate my ability to be contacted (yes, it's possible, I carefully manage my filters, teams cannot be filtered though).

I will continue to respond to pings via my username, other non-broken teams, and to my PR review dashboard.


I propose we simply set `required_reviewers: @liggitt` to be explicit about how we're (not)  handling this.

**Comment by jtslear** [comment]
Hello, @jtslear as Bug Triage Shadow for 1.19. Assigning to sig-contributor-experience. If this is incorrect, please notify us and remove the sig.

/sig contributor-experience

**Pull request closed without merging**
